cubeviz/tools/collapse_cube.py

```diff
@@ -311,187 +228,321 @@ def hide_start_end(self, dohide):
+
+            end_index = None
+
+        if start_index < 0:
+            self.ui.start_label.setStyleSheet("color: rgba(0, 0, 255, 128)")
+            self.ui.error_label.setText('Start index is out of range, setting to end point.')
+            self.ui.error_label.setVisible(True)
+
+            start_index = 0
+            
+        if start_index > len(self.wavelengths):
+            self.ui.start_label.setStyleSheet("color: rgba(255, 0, 0, 128)")
+            self.ui.error_label.setText('Start index is out of range.')
+            self.ui.error_label.setVisible(True)
+
+            start_index = None
+            
+
+        if end_index > len(self.wavelengths):
+            self.ui.end_label.setStyleSheet("color: rgba(0, 0, 255, 128)")
+            self.ui.error_label.setText('End index is out of range, setting to end point.')
+            self.ui.error_label.setVisible(True)
+
+            end_index = len(self.wavelengths)-1
+           
+        if end_index < 0:
+            self.ui.end_label.setStyleSheet("color: rgba(255, 0, 0, 128)")
+            self.ui.error_label.setText('End index is out of range.')
+            self.ui.error_label.setVisible(True)
+
+            end_index = None
+
+        if end_index < start_index:
+            self.ui.end_label.setStyleSheet("color: rgba(255, 0, 0, 128)")
+            self.ui.error_label.setText('Start index must be less than the end index.')
+            self.ui.error_label.setVisible(True)
+            return None, None
+
+        log.debug('  returning with start_index {} and end_index {}'.format(
+            start_index, end_index))
+
+        return start_index, end_index
+
+    def _calculate_callback_simple_sigma_check(self):
+        log.debug('_calculate_callback_simgple_sigma_check')
+
+        simple_sigma = self.ui.simple_sigma_input.text().strip()
+        log.debug('    simple_sigma {}'.format(simple_sigma))
+
+        try:
+            simple_sigma = float(simple_sigma)
+        except:
+            self.ui.simple_sigma_label.setStyleSheet("color: rgba(255, 0, 0, 128)")
+            self.ui.error_label.setText('Sigma must be a floating point number.')
+            self.ui.error_label.setVisible(True)
+            return None
+
+        return simple_sigma
+
+    def _calculate_callback_advanced_sigma_check(self):
+        sigma = self.ui.advanced_sigma_input.text().strip()
+        sigma_lower = self.ui.advanced_sigma_lower_input.text().strip()
+        sigma_upper = self.ui.advanced_sigma_upper_input.text().strip()
+        sigma_iters = self.ui.advanced_sigma_iters_input.text().strip()
+
+        if len(sigma) > 0:
+            try:
+                sigma = float(sigma)
+            except:
+                self.ui.advanced_sigma_label.setStyleSheet("color: rgba(255, 0, 0, 128)")
+                self.ui.error_label.setText('Sigma must be a floating point number.')
+                self.ui.error_label.setVisible(True)
+                return None
```

Comment:
I'm not that familiar with use cases here, but why do I need `Sigma` in addition to `Sigma Lower` and `Sigma Upper`? I get an error unless I provide `Sigma` in addition to the upper and lower, but it's not at all clear why that's necessary.

Reply:
This is just for convenience for the default case.  ```astropy.stats.sigma_clip``` does it this way, over-riding Sigma if separate ```Sigma_Lower``` or ```Sigma_Upper``` are provided (not sure what happens if only one is provided).

Reply:
For the purposes of this UI, shouldn't I be allowed to omit sigma entirely if I provide both upper and lower bounds?

Reply:
I think that if `sigma` has a default, then the default should automatically populated into our UI. I think the same should be true of Start Wavelength and End Wavelength, and Start Slice and End Slice, otherwise it's not immediately obvious what's going to happen when I don't enter any values.